src/broadcast.js

```diff
@@ -0,0 +1,37 @@
+import {Tensor} from './tensor.js';
+
+/**
+ * Broadcast a Tensor to a compatible shape NumPy-style.
+ * @param {Tensor} input
+ * @param {Array} newShape
+ * @return {Tensor}
+ */
+export function broadcast(input, newShape) {
+  const newRank = newShape.length;
+  if (newRank < input.rank) {
+    throw new Error(`The rank of new shape ${newRank} is invalid.`);
+  }
+  const broadcastAxes = new Array(input.rank).fill(false);
+  for (let i = 0; i < input.rank; ++i) {
+    const newAxis = newRank - i - 1;
+    const axis = input.rank - i - 1;
+    if (input.shape[axis] === 1 && newShape[newAxis] !== 1) {
+      broadcastAxes[newAxis] = true;
```

Comment:
Here `broadcastAxes[newAxis]` should be `broadcastAxes[axis]`.

Reply:
good catch, thanks @BruceDai . I fixed it in the new commit, please take a look.

Reply:
LGTM, thanks.